docs/en/stack/security/troubleshooting.asciidoc

```diff
@@ -392,6 +392,123 @@ Kerberos/SPNEGO debug logging on JVM, add following JVM system properties:
 
 For more information about JVM system properties, see {ref}/jvm-options.html[configuring JVM options].
 
+[[trb-security-saml]]
+=== Common SAML issues
+
+*Symptoms:*
+
+. Authentication in Kibana fails and the following error is printed in the
+  Elasticsearch logs:
++
+--
+[source, shell]
+-----------------------------------------------
+Cannot find any matching realm for [SamlPrepareAuthenticationRequest{realmName=null,
+assertionConsumerServiceURL=https://my.kibana.url/api/security/v1/saml}]
+-----------------------------------------------
+--
+
+. Authentication in Kibana fails and the following error is printed in the
+Elasticsearch logs:
++
+--
+[source, shell]
+-----------------------------------------------
+Cannot find metadata for entity [your:entity.id] in [metadata.xml]
+-----------------------------------------------
+--
+
+. Authentication in Kibana fails and the following error is printed in the
+Elasticsearch logs:
++
+--
+[source, shell]
+-----------------------------------------------
+unable to authenticate user [<unauthenticated-saml-user>]
+for action [cluster:admin/xpack/security/saml/authenticate]
+-----------------------------------------------
+--
+
+. Authentication in Kibana fails and the following error is printed in the
+Elasticsearch logs:
++
+--
+[source, shell]
+-----------------------------------------------
+Authentication to realm my-saml-realm failed -
+Provided SAML response is not valid for realm saml/my-saml-realm
+(Caused by ElasticsearchSecurityException[SAML Response is not a 'success' response:
+Code=urn:oasis:names:tc:SAML:2.0:status:AuthnFailed Message=null Detail=null])
+-----------------------------------------------
+--
+
+. Authentication in Kibana fails and the following error is printed in the
+Elasticsearch logs:
++
+--
+[source, shell]
+-----------------------------------------------
+The XML Signature of this SAML message cannot be validated. Please verify that the saml
+realm uses the correct SAMLmetadata file/URL for this Identity Provider
+
+-----------------------------------------------
+--
+
+. Users are unable to login with a local username/password in Kibana because SAML is
+enabled.
+
+
+*Resolution:*
+
+. This can be resolved by correcting your `sp.acs` setting in your {es} SAML
+realm configuration or by using the `xpack.security.public.*` settings in `kibana.yml`.
+{kib} either uses the `server.host` and `server.port` or the `xpack.security.public.*`
+settings to construct the SAML ACS URL and this URL needs to match exactly the URL that
+is configured in the SAML realm configuration.
+
+NOTE: The two URLs are compared as case-sensitive strings and not as canonicalized URLs.
```

Comment:
I suspect there's not quite enough hand-holding here.
If someone is reading this, then the error message itself wasn't enough info for them, and I don't think this explanation will be either.
I think, at the very least we need a walk through of something like: _(bear in mind that it's just after midnight here, so you'll probably want to rewrite 90% of this)_
- The URL in the error message is the URL that Kibana thinks it should use for the "acs" 
- The URL in the `elasticsearch.yml` is what Elasticsearch thinks Kibana will use
- If you're getting this error then those aren't the same, but the need to be (as case-sensitive strings ...)
- Often it is as simple as changing the config in elasticsearch to match the ASC URL that Kibana is using (the one from the error message).
- However, if you think that the URL that Kibana is sending is wrong (e.g. Kibana is behind a proxy/load balancer with a different URL, or is accessed via a custom DNS name), then you can reconfigure Kibana so that it is aware of the correct URL:
    1. you can change `server.host` and `server.port` to cause Kibana to listen on a different URL. This may affect the URLs that users need to use to access Kibana, so you should consult the Kibana documentation for more details.
    2. You can configure Kibana to be aware that its _public_ URL is not the same as the server and port it listens on. The `xpack.security.public.*` settings are used to construct the ACS URL without changing the server and port that Kibana listens on.
- In either case, bear in mind that your IdP needs to be configured with this URL as well, so whatever URL you end up with in your `elasticsearch.yml` will need to match what you put in your IdP.